### PR TITLE
PR: Update to Node v12

### PR DESCRIPTION
**Pull request by nelsonic**
This pull request: 
+ [x] improves code maintainability by reducing complexity in several functions #63 
+ [x] update all `devDependencies` to latest versions #64 
+ [x] updates the package to use the latest version of Node.js (v12) #65 
+ [x] switch to Travis-CI from CodeShip for CI #67 
+ [x] replace `Mocha` with `Tape` for reliability! #68

### CODE REVIEW

**Comment by nelsonic** [comment]
Having the PR build _pass_ but the branch build _fail_ is getting old ... 😞 
image

**Comment by nelsonic** [comment]
Hi @iteles 👋 
This is a pretty big PR because there was quite a lot of code to update.
I changed the test framework from Mocha to Tape for speed and consistency.
The biggest addition is the `package-lock.json` which I think makes sense in this project from a debugging perspective. If anyone `else` can't get it to work at least we know they are using the _exact_ same version of the `aws-sdk` as us if we need to help them debug _why_. 💭 

Please review & merge when you can.
Thanks! ✨

**Comment by nelsonic** [comment]
I've been using this successfully for the past couple of days:
image

Deployed a couple of Lambda Functions:
aws-lambda-functions-deployed

The only reason the PR took a bit longer is that I was updating some of the code and tests. 
But it still works _exactly_ the way it did before (5 years ago!) 👍

File: README.md [hunk 1]
```diff
@@ -217,39 +244,11 @@ https://github.com/dwyl/aws-lambda-deploy/issues/33
 
 <br />
 
-## tl;dr
```
Comment (iteles) [inline README.md]: :+1:  Agreed this is now less relevant and cuts the flow a little too much.

File: README.md [hunk 2]
```diff
@@ -153,13 +175,18 @@ Example:
...
-Check our `package.json` if in doubt.
+Check our
+[package.json#L14-L17](https://github.com/dwyl/aws-lambda-deploy/blob/81766f9c20157039e14703e36dbbbaef4cfb4ac3/package.json#L14-L17)
+file if in doubt.
 
-## Explanation of the Steps
+<br />
+
+
+## Explanation of the Steps 👩‍🏫
```
Comment (iteles) [inline README.md]: Love a good emoji 😄

File: lib/copyfiles.js [hunk 3]
```diff
@@ -19,42 +19,27 @@ function ensureEnv (filelist) {
...
-      var dest = destination + fd.replace(basepath, '');
-      var contents;
-      if (!fd.match(/\.json/) && !fd.match(/\.env/) && babelOpts) { // only transpile ES6
-        var transpiled = babel.transformFileSync(fd, babelOpts); // from .babelrc
-        contents = transpiled.code;
-      } else {
-        contents = fs.readFileSync(fd, 'utf8');
-      }
-      fs.writeFileSync(dest, contents);               // write sync.
+      const dest = destination + fd.replace(basepath, '');
```
Comment (iteles) [inline lib/copyfiles.js]: Way simpler 👍

File: README.md [hunk 4]
```diff
@@ -1,67 +1,83 @@
...
 
-Deploy Amazon Web Services Lambda function(s) with a single command.
+Deploy your AWS Lambda function(s) in seconds with a single command.
 
-[![Codeship Build Status](https://img.shields.io/codeship/cb362fc0-b8a0-0133-b733-0e8881fc1b37.svg?maxAge=2592000?style=flat-square)](https://www.npmjs.com/package/dpl)
-[![codecov.io](https://img.shields.io/codecov/c/github/dwyl/aws-lambda-deploy.svg?maxAge=2592000?style=flat-square)](https://codecov.io/github/dwyl/aws-lambda-deploy?branch=master)
-[![Code Climate](https://img.shields.io/codeclimate/github/dwyl/aws-lambda-deploy.svg?maxAge=2592000?style=flat-square)](https://codeclimate.com/github/dwyl/aws-lambda-deploy)
+[![Build Status](https://img.shields.io/travis/dwyl/aws-lambda-deploy/master.svg?style=flat-square)](https://travis-ci.org/dwyl/aws-lambda-deploy)
+[![codecov.io](https://img.shields.io/codecov/c/github/dwyl/aws-lambda-deploy/master.svg?style=flat-square)](http://codecov.io/github/dwyl/aws-lambda-deploy?branch=master)
+[![Code Climate maintainability](https://img.shields.io/codeclimate/maintainability/dwyl/aws-lambda-deploy?color=brightgreen&style=flat-square)](https://codeclimate.com/github/dwyl/aws-lambda-deploy)
```
[the remark this answers is not in the document]
Comment (nelsonic) [reply README.md]: @iteles the rich diff is not relevant given that I've completely updated the badges. see:
https://github.com/dwyl/aws-lambda-deploy/tree/update-to-node-v12-issue%2365
image

Once http://www.github.com/dwyl/hits/issues/93 is sorted the badges will be fine. 👍

**Comment by nelsonic** [comment]
@iteles this PR _resolves_ the old Badges.
Thanks for reviewing. Please merge when OK. 👍

**Pull request merged**